packages/runtime-dom/src/directives/vShow.ts

```diff
@@ -20,15 +20,18 @@ export const vShow: ObjectDirective<VShowElement> = {
     }
   },
   updated(el, { value, oldValue }, { transition }) {
-    if (transition && value !== oldValue) {
-      if (value) {
-        transition.beforeEnter(el)
-        setDisplay(el, true)
-        transition.enter(el)
-      } else {
-        transition.leave(el, () => {
-          setDisplay(el, false)
-        })
+    if (!value === !oldValue) return
+    if (transition) {
+      if (value !== oldValue) {
```

Comment:
Do we still need this `if`? Shouldn't `value` be always different from `oldValue` here?

Reply:
`updated` is run whenever the *component* has updated, not when the value actually changed.

So yes, we need it.

Reply:
What I mean, we are returning if `!value === !oldValue`, meaning that here, `value !== oldValue` will be always `true`. Or am I missing something?

Reply:
Ah! Missed that.